src/Application.vala

```diff
@@ -216,15 +220,15 @@ public class Terminal.Application : Gtk.Application {
     protected override int command_line (ApplicationCommandLine command_line) {
         unowned var options = command_line.get_options_dict ();
         var window = (MainWindow) active_window;
-        bool new_window;
+        bool new_window = false;
+        options.lookup ("new-window", "b", out new_window);
+        /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
+         * Currently they are set to restore (subject to the restore-tabs setting).
+         * If it is desired that tabs should never be restored in these circimstances add another check below.
+         */
```

Comment:
this commentary doesn't make sense any more. Before, launching with one of `-x`, `-e`, or with a command didn't restored tabs. But now the behaviour is to always restore if `-n` is not used too.